lib/core/ingestion/ingester.rb

```diff
@@ -11,8 +11,10 @@
 module Core
   module Ingestion
     class Ingester
-      def initialize(sink_strategy)
+      def initialize(sink_strategy, max_allowed_document_size = 5 * 1024 * 1024)
```

Comment:
Will this threshold be related to the queue size? Are you planning to allow them configured separately?

Reply:
It's a part of the mentioned follow-up:

> Make sure that maximum allowed document size is always less than maximum queue volume

For first iteration I believe they can just be the same. If later we'll need different numbers, we can allow for it, but will we ever actually need it? Probably not.